### DISCUSSION TITLE
[Feature Request] Support for accountless, offline achievements

### DISCUSSION THREAD

**Discussion by dio-gh** [RetroAchievements/RAWeb#1043] [Ideas]
Hi all,

I've been wondering if it was possible to facilitate support for accountless, offline achievements, and if you guys would be okay with that in the first place. The core idea is that some people (including me) may not be interested in tracking/sharing their achievements online, at least not right away, and thus might find having to create a RetroAchievements account an unnecessary drag, or might not even discover the feature in the first place¹.

For this to work, emulators that implement RetroAchievements support would need to be able to either scrape² all the achievements, or collect a designated archive automatically prepared for this purpose. This could be some agreed upon archive containing a cheat database with all the assets (icons, names, descriptions) in some agreed upon format, or something of the sort.

What do you guys reckon?

¹ People may or may not know about RetroAchievements, and the settings panels of emulators is not usually the most accessible thing. If they never get an achievement, they might not know it's possible to get any, especially not that it needs to be set up.

² While it is technically already possible to scrape the achievements through the website, what I'd be thinking of is more in the way of a new API endpoint designated for this purpose (supplying arbitrary achievement data). Hence why I'm opening the ticket here.

P.S.: Apologies if I'm posting this to the wrong repo, I'm not familiar with this project beyond superficial levels.

**Comment by Jamiras** [comment]
The [rcheevos APIs](https://github.com/RetroAchievements/rcheevos/wiki/rc_api_process_fetch_game_data_response) don't really care where the data comes from. You could write your own version of an emulator that reads the data from local files instead of fetching them from the server. I don't think it's something we would ever do, as the constant influx of new achievements (and achievement fixes) have to be centralized somewhere.

An alternative for you would be to create a local instance of the server and connect to that from your clients. We even have a sanitized copy of the database from a couple years ago you can initialize your local site with: https://github.com/RetroAchievements/RAWeb/discussions/853#discussioncomment-1671098

**Comment by meleu** [comment]
just trying to clarify:

Your request seems to be just "provide an unauthenticated endpoint to get achievements data". At least that's what is up to the code in this repo...

The "offline achievements" feature needs to be handled in the emulator's side. Probably requiring some work in the rcheevos code.
